tools/schemapi/utils.py

```diff
@@ -363,53 +372,47 @@ def title(self) -> str:
         else:
             return ""
 
+    @overload
+    def get_python_type_representation(
+        self,
+        for_type_hints: bool = ...,
+        return_as_str: Literal[True] = ...,
+        additional_type_hints: list[str] | None = ...,
+    ) -> str: ...
+    @overload
+    def get_python_type_representation(
+        self,
+        for_type_hints: bool = ...,
+        return_as_str: Literal[False] = ...,
+        additional_type_hints: list[str] | None = ...,
+    ) -> list[str]: ...
     def get_python_type_representation(
         self,
         for_type_hints: bool = False,
-        altair_classes_prefix: str | None = None,
```

Comment:
Tried to figure it out myself but I'm a bit lost. Where does the code generation now add the `core` prefix?

Reply:
> Tried to figure it out myself but I'm a bit lost. Where does the code generation now add the `core` prefix?

It doesn't have to 😎 

Since we're using `from __future__ import annotations`, a lot of that code wasn't needed anymore.
`core` specifically was removed in the only 2 cases it appeared:

https://github.com/vega/altair/blob/15f115134e6a6d3234ccc7230f65b1fb1d10f8c8/tools/schemapi/utils.py#L406-L412

Which was the major source of reducing code in https://github.com/vega/altair/pull/3431

---

So I removed a lot of code that hadn't had an effect since then.
I tried to demonstrate that by pushing this commit [`0014cc8` (#3515)](https://github.com/vega/altair/pull/3515/commits/0014cc8e5527f8115c4cd2f4cef9e3e382cd4379) which was the one prior to this

Reply:
Nice! :D